website/views.py

```diff
@@ -207,11 +208,15 @@ def forgot_password_form():
 
 def resolve_guid_download(guid, provider=None):
     try:
-        guid = Guid.objects.get(_id=guid.lower())
+        if VersionedGuidMixin.GUID_VERSION_DELIMITER in guid:
+            base_guid, version = guid.split(VersionedGuidMixin.GUID_VERSION_DELIMITER)
+            base_guid_obj = Guid.objects.get(_id=base_guid.lower())
+        else:
+            base_guid_obj = Guid.objects.get(_id=guid.lower())
     except Guid.DoesNotExist:
         raise HTTPError(http_status.HTTP_404_NOT_FOUND)
 
-    resource = guid.referent
+    resource = base_guid_obj.referent
```

Comment:
Similar case here, we need to get the version referent if version exists.

```python
# TODO: turn this into a helper method `get_referent_from_versioned_guid()`
guid_split_list = guid.lower().split(VersionedGuidMixin.GUID_VERSION_DELIMITER)
guid = guid_split_list[0]
version = guid_split_list[1] if len(guid_split_list) > 1 else None
if version:
    resource = GuidVersionsThrough.objects.get(guid___id=guid, version= version).referent
else:
    resource = Guid.objects.get(_id=guid).referent
```

Reply:
Changed it